python_modules/dagster/dagster/_utils/error.py

```diff
@@ -94,14 +138,29 @@ def serializable_error_info_from_exc_info(
     exc_type = check.not_none(exc_type, additional_message=additional_message)
     e = check.not_none(e, additional_message=additional_message)
     tb = check.not_none(tb, additional_message=additional_message)
-    from dagster._core.errors import DagsterUserCodeProcessError
+
+    from dagster._core.errors import DagsterUserCodeExecutionError, DagsterUserCodeProcessError
+
+    if isinstance(e, DagsterUserCodeExecutionError) and _should_redact_user_code_error():
+        masked_tb_exc = _get_masked_traceback_exception(with_instruction=True)
+        print(  # noqa: T201
+            _serializable_error_info_from_tb(
+                _get_masked_traceback_exception(with_instruction=False)
+            ).to_string(),
+            file=sys.stderr,
+        )
```

Comment:
* can do sys.stderr.write to avoid print 
* not clear we need to print the redacted exception wrapping - maybe just an explicit string prefixing before writing out the real error
* not sure what the right way to do this is so maybe out of scope for initial version, but we probably need to respect log setting in some capacity of someone is for example hoping to find via structured json ingesting in to log capture system

Reply:
Updated to just print the underlying error with a line preceding that holds the error ID.